go/vt/wrangler/reparent.go

```diff
@@ -944,23 +945,25 @@ func (wr *Wrangler) emergencyReparentShardLocked(ctx context.Context, ev *events
 
 	// Verify masterElect is alive and has the most advanced position
 	masterElectStatus, ok := statusMap[masterElectTabletAliasStr]
-	if !ok {
+	if !ok || slaveWasRunning(masterElectStatus) && masterElectStatus.After == nil {
```

Comment:
What's the reason for adding this new check?

Reply:
It's just an extra layer of safety. We should theoretically never hit it. It's just a bit of defensive programming in case code changes in the future that would cause us not to get an error back, for whatever reason.

Reply:
I think it's cleaner in general to establish reasonable contracts, document them, and then rely on them. Code would be very ugly if we distrusted documented contracts. In this case, I think a reasonable contract for the RPC is that it should return a non-nil error if it was unable to determine the status.

Reply:
Removed.